File: desert/shine-getter.lua
function pos2xy(pos) return pos['x'], pos['y'] end

function bestCoin(xs)
    local r, maxR = nil, 0
    for i = 1, #xs do
        local v = xs[i].value/self:distanceTo(xs[i])
        if v > maxR then
            r, maxR = xs[i], v
        end
    end
    return r
end

loop
    local c = bestCoin(self:findItems())
    self:moveXY(pos2xy(c.pos))
end

File: desert/shine-getter.clj
(while true
  (def is (.findItems this))
  (def c nil)
  (when (seq is)
    (def cv 0)
    (def i 0)
    (while (< i (count is))
      (if (> (/ (.value (nth is i)) (.distanceTo this (nth is i))) cv)
        (when true
          (def c (nth is i))
          (def cv (/ (.value (nth is i)) (.distanceTo this (nth is i))))))
      (def i (inc i))))
  (if c
    (.move this (.pos c))
    (.wait this 0.1)))
